File: scalaz/src/main/scala/org/openapitools/client/api/RequestRoutesDepartureSearch.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import org.joda.time.DateTime

import RequestRoutesDepartureSearch._

case class RequestRoutesDepartureSearch (
  id: String,
departureLocationId: String,
arrivalLocationIds: List[String],
transportation: RequestTransportation,
departureTime: DateTime,
properties: List[RequestRoutesProperty],
range: Option[RequestRangeFull])

object RequestRoutesDepartureSearch {
  import DateTimeCodecs._

  implicit val RequestRoutesDepartureSearchCodecJson: CodecJson[RequestRoutesDepartureSearch] = CodecJson.derive[RequestRoutesDepartureSearch]
  implicit val RequestRoutesDepartureSearchDecoder: EntityDecoder[RequestRoutesDepartureSearch] = jsonOf[RequestRoutesDepartureSearch]
  implicit val RequestRoutesDepartureSearchEncoder: EntityEncoder[RequestRoutesDepartureSearch] = jsonEncoderOf[RequestRoutesDepartureSearch]
}

File: fsharp-functions/OpenAPI/src/model/RequestRoutesDepartureSearch.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.RequestRangeFull
open OpenAPI.Model.RequestRoutesProperty
open OpenAPI.Model.RequestTransportation

module RequestRoutesDepartureSearch = 

  //#region RequestRoutesDepartureSearch

  [<CLIMutable>]
  type RequestRoutesDepartureSearch = {
    [<JsonProperty(PropertyName = "id")>]
    Id : string;
    [<JsonProperty(PropertyName = "departure_location_id")>]
    DepartureLocationId : string;
    [<JsonProperty(PropertyName = "arrival_location_ids")>]
    ArrivalLocationIds : string[];
    [<JsonProperty(PropertyName = "transportation")>]
    Transportation : RequestTransportation;
    [<JsonProperty(PropertyName = "departure_time")>]
    DepartureTime : DateTime;
    [<JsonProperty(PropertyName = "properties")>]
    Properties : RequestRoutesProperty[];
    [<JsonProperty(PropertyName = "range")>]
    Range : RequestRangeFull;
  }
  
  //#endregion
  